pkg/apis/etcd/v1alpha1/types.go

```diff
@@ -23,6 +23,12 @@ type EtcdStorageConditionType string
 const (
 	// Deployed means EtcdProxy ReplicaSet and Service for exposing EtcdProxy are created.
 	Deployed EtcdStorageConditionType = "Deployed"
+
+	// ReplicaSetFailure means ReplicaSet for etcdproxy has not been created successfully.
+	ReplicaSetFailure EtcdStorageConditionType = "ReplicaSetFailure"
```

Comment:
I believe names could be improved. I'll try to think of better ones. If you have ideas, let me know. :)